rainier-core/src/main/scala/com/stripe/rainier/core/Discrete.scala

```diff
@@ -95,3 +167,54 @@ final case class BetaBinomial(a: Real, b: Real, k: Real) extends Discrete {
       Combinatorics.beta(a + v, k - v + b) -
       Combinatorics.beta(a, b)
 }
+
+/**
+  * Discrete Mixture Distribution
+  *
+  * @param components Map of Discrete distribution and probabilities
+  */
+final case class DiscreteMixture(components: Map[Discrete, Real])
+    extends Discrete {
+  val generator: Generator[Int] =
+    Categorical(components).generator.flatMap { d =>
+      d.generator
+    }
+
+  def logDensity(v: Real): Real =
+    Real
+      .logSumExp(components.map {
+        case (dist, weight) => {
+          dist.logDensity(v) + weight.log
+        }
+      })
+}
+
+/**
+  * Zero Inflated Poisson with expecation `psi*lambda`
+  *
+  * @param psi    The probability of non-zero count
+  * @param lambda The expected (possibly non-zero) count value
+  */
+final case class ZeroInflatedPoisson(psi: Real, lambda: Real) extends Discrete {
```

Comment:
Rather than having special classes for these, why not just have a `zeroInflated(psi: Real)` combinator on `Discrete`?

Reply:
Perfect. How about I add `constantInflated(constant: Real, psi: Real)`? Also, would you be ok if I change the threshold in the unit tests from within 5% to 10% for the expected values? Below, I changed the sampling from `1000` to `10000` to get the convergence. It might be easier to keep `1000` to keep the tests from taking too long.

Reply:
Yep, sounds good on all counts. (Though I wonder if `zeroInflated` is enough of a special case to deserve its own convenience method.)